homeassistant/components/hunterdouglas_powerview/cover.py

```diff
@@ -192,29 +222,54 @@ async def async_stop_cover(self, **kwargs):
         self._async_update_from_command(await self._shade.stop())
         await self._async_force_refresh_state()
 
+    @callback
+    def _clamp_cover_limit(self, target_hass_position):
+        """Dont allow a cover to go into an impossbile position."""
+        # no override required in base
+        return target_hass_position
+
     async def async_set_cover_position(self, **kwargs):
         """Move the shade to a specific position."""
         if ATTR_POSITION not in kwargs:
             return
-        await self._async_move(kwargs[ATTR_POSITION])
+        await self._async_move(self._clamp_cover_limit(kwargs[ATTR_POSITION]))
+
+    @callback
+    def _set_shade_postion(self, target_hass_position):
+        position_one = hass_position_to_hd(target_hass_position)
+        self.set_position_primary(position_one)
```

Comment:
We need to handle the case where the async_move call fails.  We shouldn't update the internal position until we get a success from the api.

I'll take a go at cleaning this up after lunch

Reply:
https://github.com/home-assistant/core/pull/62788#issuecomment-1140315512